### PR TITLE
Add task/buildpacks-phases/0.2

### PR DESCRIPTION
**Pull request by jromero**
# Changes

- Add `APP_IMAGE_DIGEST` result
- Upgrade to Platform API 0.4
- Add parameter to configure the `LIFECYCLE_IMAGE`
- Add support for build-time environment variables
- Replace use of PipelineResources with Params

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality of task changed or new task added)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)
- [x] Complies with [Catalog Organization TEP][TEP], see [example]. **Note** [An issue has been filed to automate this validation][validation]
  - [x] File path follows  `<kind>/<name>/<version>/name.yaml`
  - [x] Has `README.md` at `<kind>/<name>/<version>/README.md`
  - [x] Has mandatory `metadata.labels` - `app.kubernetes.io/version` the same as the `<version>` of the resource
  - [x] Has mandatory `metadata.annotations` `tekton.dev/pipelines.minVersion`
  - [x] mandatory `spec.description` follows the convention

          ```

          spec:
            description: >-
              one line summary of the resource

              Paragraph(s) to describe the resource.
          ```

_See [the contribution guide](https://github.com/tektoncd/catalog/blob/master/CONTRIBUTING.md)
for more details._

---

[TEP]: https://github.com/tektoncd/community/blob/master/teps/0003-tekton-catalog-organization.md
[example]: https://github.com/tektoncd/catalog/tree/master/task/git-clone/0.1
[validation]:  https://github.com/tektoncd/catalog/issues/413

### CODE REVIEW

File: task/buildpacks-phases/0.2/README.md
Comment (vinamra28) [inline task/buildpacks-phases/0.2/README.md:15]: I see the cache directory is `optional` so it requires min Tekton version to be v0.17.0`

File: task/buildpacks-phases/0.2/buildpacks-phases.yaml
Comment (vinamra28) [inline task/buildpacks-phases/0.2/buildpacks-phases.yaml:9]: same as above min version needs to be `0.17.0`

**Comment by jromero** [comment]
Thank you @vinamra28 - comments have been addressed.

**Review by vinamra28 — APPROVED** [review]
/meow

**Comment by jromero** [comment]
Thanks @vinamra28. What else is necessary here? Who should be adding the lgtm label?

**Comment by vinamra28** [comment]
Hi @jromero, just waiting for another set of eyes for review 🙂

**Comment by chmouel** [comment]
Thanks!

the diff to make it easier to review https://paste.gg/p/anonymous/dccee339d8f94184a43bb5c3bcf3fd4d#paste-diff-l126

/lgtm

**Pull request merged**
